src/trie/trie.c

```diff
@@ -80,21 +80,48 @@ TrieNode *__newTrieNode(const rune *str, t_len offset, t_len len, const char *pa
   return n;
 }
 
+TrieNode *__trieNode_resizeChildren(TrieNode *n, int offset) {
+  n = rm_realloc(n, __trieNode_Sizeof(n->numChildren + offset, n->len));
+  TrieNode **children = __trieNode_children(n);
+
+  // stretch or shrink the child key cache array
+  memmove(((rune *)children) + offset, (rune *)children, sizeof(TrieNode *) * n->numChildren);
```

Comment:
Why you need to cast it to rune?

Reply:
It is more explicit.
We do the same on https://github.com/RediSearch/RediSearch/blob/4d141cc6282b5acdd2bf695601161ab129828d14/deps/triemap/triemap.c#L38